### PR TITLE
IGNITE-22669 Change maven.compiler.source to 11

### PR DESCRIPTION
**Pull request by timoninmaxim**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

**Comment by alex-plekhanov** [comment]
If it's a final patch for moving to java 11 we should also:
- Change ./bin files to check correct java version (11 instead of 8, for example ignite.bat/functions.sh)
- Remove java-8 profile from tools/pom.xml
- Remove java8 directory from tools module
- Change JDK verrsion in all Dockerfiles
- Change JDK in deb/rpm packages
- Perhaps clean some code for old java versions (for example, GridCleaner.PRE_JAVA9_CLEANER_CLASS_NAME)
- Fix cpp/dotnet platforms to use java 11
- ...

**Comment by nizhikov** [comment]
IMAGE 2024-09-03 16:28:44
IMAGE 2024-09-03 16:28:48
IMAGE 2024-09-03 16:28:51

We still have some links to the older java docs. 
Do we want to fix them?

File: modules/compatibility/src/test/java/org/apache/ignite/compatibility/persistence/PersistenceBasicCompatibilityTest.java [hunk 1]
```diff
@@ -108,7 +108,7 @@ public void testNodeStartByOldVersionPersistenceData() throws Exception {
             Assume.assumeTrue("Skipped on jdk " + U.jdkVersion(),
                 VER_2_12_0.compareTo(IgniteReleasedVersion.fromString(version)) < 0);
         }
-        else if (majorJavaVer > 8) {
+        else if (majorJavaVer == 11) {
```
Comment (nizhikov) [inline modules/compatibility/src/test/java/org/apache/ignite/compatibility/persistence/PersistenceBasicCompatibilityTest.java:111]: What is this test will be executed on 12, 13 or any other non LTS jdk release?

**Review by nizhikov — APPROVED** [review]

File: modules/compatibility/src/test/java/org/apache/ignite/compatibility/persistence/PersistenceBasicCompatibilityTest.java (hunk 1 above)
Comment (nizhikov) [reply modules/compatibility/src/test/java/org/apache/ignite/compatibility/persistence/PersistenceBasicCompatibilityTest.java:111]: Nevermind, case for a jdk > 11 is few lines higher.

**Pull request merged**